src/Linux/Avalonia.LinuxFramebuffer/Output/DrmOutput.cs

```diff
@@ -15,20 +16,39 @@ namespace Avalonia.LinuxFramebuffer.Output
 {
     public unsafe class DrmOutput : IGlOutputBackend, IGlPlatformSurface
     {
+        private DrmOutputOptions _outputOptions = new();
         private DrmCard _card;
         public PixelSize PixelSize => _mode.Resolution;
-        public double Scaling { get; set; }
+
+        public double Scaling
+        {
+            get => _outputOptions.Scaling;
+            set => _outputOptions.Scaling = value;
+        }
         public IGlContext PrimaryContext => _deferredContext;
 
         private EglPlatformOpenGlInterface _platformGl;
         public IPlatformOpenGlInterface PlatformOpenGlInterface => _platformGl;
 
-        public DrmOutput(string path = null)
+        public DrmOutput(DrmCard card, DrmResources resources, DrmConnector connector, DrmModeInfo modeInfo,
+            DrmOutputOptions? options = null)
         {
-            var card = new DrmCard(path);
+            if(options != null) 
+                _outputOptions = options;
+            Init(card, resources, connector, modeInfo);
+        }
+        public DrmOutput(string path = null, bool connectorsForceProbe = false, [CanBeNull] DrmOutputOptions options = null)
+        {
+            if(options != null) 
+                _outputOptions = options;
+            CreateDrmOutput(path, connectorsForceProbe);
+        }
 
-            var resources = card.GetResources();
+        private void CreateDrmOutput(string path = null, bool connectorsForceProbe = false)
+        {
```

Comment:
I somehow feels that you can safely remove those 4 lines

Reply:
which lines do you mean exactly?

Reply:
44-48

Reply:
ah now I get it :)